tests/test_voting.py

```diff
@@ -0,0 +1,203 @@
+import pytest
+from collections import Counter
+
+from bigchaindb.core import Bigchain
+from bigchaindb.voting import Voting, INVALID, VALID, UNDECIDED
+
+
+################################################################################
+# Tests for checking vote eligibility
+
+
+def test_partition_eligible_votes():
+    class TestVoting(Voting):
+        @classmethod
+        def verify_vote_signature(cls, vote):
+            if vote['node_pubkey'] == 'invalid sig':
+                return False
+            if vote['node_pubkey'] == 'value error':
+                raise ValueError()
+            return True
+
+    voters = ['valid', 'invalid sig', 'value error', 'not in set']
+    votes = [{'node_pubkey': k} for k in voters]
+
+    el, inel = TestVoting.partition_eligible_votes(votes, voters[:-1])
+    assert el == [votes[0]]
+    assert inel == votes[1:]
+
+
+################################################################################
+# Test vote counting
+
+
+def test_count_votes():
+    class TestVoting(Voting):
+        @classmethod
+        def verify_vote_schema(cls, vote):
+            return vote['node_pubkey'] != 'malformed'
+
+    voters = (['cheat', 'cheat', 'says invalid', 'malformed'] +
+              ['kosher' + str(i) for i in range(10)])
+
+    votes = [Bigchain(v).vote('block', 'a', True) for v in voters]
+    votes[2]['vote']['is_block_valid'] = False
+    # Incorrect previous block subtracts from n_valid and adds to n_invalid
+    votes[-1]['vote']['previous_block'] = 'z'
+
+    assert TestVoting.count_votes(votes) == {
+        'counts': {
+            'n_valid': 9,    # 9 kosher votes
+            'n_invalid': 4,  # 1 cheat, 1 invalid, 1 malformed, 1 rogue prev block
```

Comment:
> As far as I understand this is because a node voted twice and only one vote is considered

That's right. I added a comment to the test to make it more clear, hope that's ok.